up-transport-vsomeip/src/transport_inner.rs

```diff
@@ -0,0 +1,726 @@
+/********************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+use crate::determine_message_type::RegistrationType;
+use crate::listener_registry::{add_client_id_app_name, find_app_name};
+use crate::message_conversions::convert_umsg_to_vsomeip_msg_and_send;
+use crate::vsomeip_offered_requested::{
+    insert_event_requested, insert_service_offered, insert_service_requested, is_event_requested,
+    is_service_offered, is_service_requested,
+};
+use crate::{split_u32_to_u16, ApplicationName, ClientId};
+use cxx::{let_cxx_string, UniquePtr};
+use log::{error, info, trace};
+use std::path::{Path, PathBuf};
+use std::thread;
+use std::time::Duration;
+use tokio::runtime::Builder;
+use tokio::sync::mpsc::{channel, Receiver, Sender};
+use tokio::sync::oneshot;
+use up_rust::{UCode, UMessage, UMessageType, UStatus, UUri};
+use vsomeip_sys::extern_callback_wrappers::MessageHandlerFnPtr;
+use vsomeip_sys::glue::{
+    make_application_wrapper, make_runtime_wrapper, ApplicationWrapper, RuntimeWrapper,
+};
+use vsomeip_sys::safe_glue::{
+    get_pinned_application, get_pinned_runtime, register_message_handler_fn_ptr_safe,
+    request_single_event_safe,
+};
+use vsomeip_sys::vsomeip;
+use vsomeip_sys::vsomeip::{ANY_MAJOR, ANY_MINOR};
+
+pub const UP_CLIENT_VSOMEIP_TAG: &str = "UPClientVsomeipInner";
+pub const UP_CLIENT_VSOMEIP_FN_TAG_APP_EVENT_LOOP: &str = "app_event_loop";
+pub const UP_CLIENT_VSOMEIP_FN_TAG_REGISTER_LISTENER_INTERNAL: &str = "register_listener_internal";
+pub const UP_CLIENT_VSOMEIP_FN_TAG_UNREGISTER_LISTENER_INTERNAL: &str =
+    "unregister_listener_internal";
+pub const UP_CLIENT_VSOMEIP_FN_TAG_SEND_INTERNAL: &str = "send_internal";
+pub const UP_CLIENT_VSOMEIP_FN_TAG_INITIALIZE_NEW_APP_INTERNAL: &str =
+    "initialize_new_app_internal";
+pub const UP_CLIENT_VSOMEIP_FN_TAG_START_APP: &str = "start_app";
```

Comment:
do all these need to be public?

Reply:
In the context of being usable within transport_inner, yes. But some of this is resolved by changing the structure of the modules that just went into this PR branch.